packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -301,7 +301,7 @@ export function createHooksInternal<
             (...args: any[]) => {
               const [queryKey, ...rest] = args;
               return queryClient.invalidateQueries(
-                getArrayQueryKey(queryKey),
+                getArrayQueryKey(queryKey, 'any'),
```

Comment:
No option to specify either `query` or `infinite` here ... Will invalidate indiscriminately!

Reply:
Can we add a test to make sure this invalidates both queries and infinite queries?

Reply:
Yep! I have tested it and it deffinitely does so but I never added a test 🙈